src/sonic-yang-models/yang-models/sonic-tunnel.yang

```diff
@@ -0,0 +1,117 @@
+module sonic-tunnel {
+    yang-version 1.1;
+    namespace "http://github.com/Azure/sonic-tunnel";
+    prefix tunnel;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+    
+    import sonic-types {
+        prefix stypes;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONiC DualToR tunnel data";
+
+    revision 2022-08-23 {
+        description
+            "Initial revision";
+    }
+
+    container sonic-tunnel {
+        container TUNNEL {
+            list TUNNEL_LIST {
+                max-elements 1;
+
+                key "mux_tunnel";
+                
+                leaf mux_tunnel {
+                    description "Name of MuxTunnel";
+                    type string {
+                        pattern "MuxTunnel[0-9]+";
+                    }
+                }
+
+                leaf dscp_mode {
+                    description "DSCP mode";
+                    type string {
+                        pattern "uniform|pipe";
+                    }
+                }
+
+                leaf src_ip {
+                    type inet:ipv4-address;
+                    description "optional source IPv4 address.";
+                }
+
+                leaf dst_ip {
+                    type inet:ipv4-address;
+                    description "SONiC DualToR peer IPv4 address.";
+                }
+
+                leaf ecn_mode {
+                    description "ECN mode";
+                    type string {
+                        pattern "copy_from_outer|standard";
+                    }
+                }
+
+                leaf encap_ecn_mode {
+                    description "Encap ECN mode";
+                    type string {
+                        pattern "standard";
+                    }
+                }
+
+                leaf ttl_mode {
+                    description "TTL mode";
+                    type string {
+                        pattern "uniform|pipe";
+                    }
+                }
+
+                leaf tunnel_type {
+                    description "Tunnel type";
+                    type string {
+                        pattern "IPINIP";
+                    }
+                }
+
+                leaf decap_dscp_to_tc_map {
+                    description "optional Decap DSCP to TC map";
+                    type string {
+                        pattern "AZURE_TUNNEL";
```

Comment:
only 1 pattern needed?

Reply:
As far as I can tell this was the only configuration supported, Is there a different way I should specify this?